Add bundled IO support to portable executor runner

mansnils opened the pull request:
Executor runner supports both models with/wo bundled io in same path.
To enable bundled IO EXECUTORCH_BUILD_DEVTOOLS and
EXECUTORCH_ENABLE_BUNDLE_IO are required.

Adds tests in Arm backend for testing this/depending on this.
Except for enabling bundle-io for VGF backend where applicable,
some additional resnets model tests are enabled as well.

Avoids narrowing conversion errors in pte_to_header script by switching
char to unsigned char.

cc @freddan80 @per @zingo @oscarandersson8218 @digantdesai

mansnils committed:
Add bundled IO support to portable executor runner

Executor runner supports both models with/wo bundled io in same path.
To enable bundled IO EXECUTORCH_BUILD_DEVTOOLS and
EXECUTORCH_ENABLE_BUNDLE_IO are required.

Adds tests in Arm backend for testing this/depending on this.
Except for enabling bundle-io for VGF backend where applicable,
some additional resnets model tests are enabled as well.

Avoids narrowing conversion errors in pte_to_header script by switching
char to unsigned char.

Signed-off-by: Måns Nilsson <mans.nilsson@arm.com>
Change-Id: Iffdf6c253a57b01abb02758b6ba2d1e3de25e3d8

JacobSzwejbka commented on examples/portable/executor_runner/executor_runner.cpp:
```diff
@@ -460,5 +560,52 @@ int main(int argc, char** argv) {
   }
 
+  if (bundle_io) {
+    // With bundled io we can check the result.
```
This file gets copy pasted a ton as a basic example of a runner. Im pretty resistant to something like this which will tempt people to use it it as a tester. Im already not a huge fan of having the benchmarking stuff in here.

mansnils commented on examples/portable/executor_runner/executor_runner.cpp:
```diff
@@ -92,6 +105,17 @@ using executorch::runtime::Span;
 using executorch::runtime::Tag;
 using executorch::runtime::TensorInfo;
 
+#if defined(ET_BUNDLE_IO_ATOL)
```
Will do

mansnils replied:
Done

mansnils committed:
Fix buck build error

Change-Id: I91ca1425bc023a37656704b3bae82da26cbd58e4

mansnils committed:
Merge remote-tracking branch origin/upstream/main

mansnils committed:
Fix inputs lifetime issue in executor runner

Also slightly change condition in order to avoid confusing print,
'Loading inputs from input file(s)' when no files where provided.

Change-Id: Icc9c1faa50a1c00e0a6ccf12c85b687af736f6b2

mansnils committed:
Provide optional bundled io atol/rtol as parameters

Change-Id: I80148fffc4b25e5470b53b559e184ac08630426a

mansnils committed:
Merge remote-tracking branch origin/upstream/main

mansnils committed:
Correct input handling in executor runner

Change-Id: I2c3e7e25f48d56071861218403941c451baf3a69

mansnils commented:
> @mansnils Do you have strong preference to not fork the runner?
> 
> We currently have a slight preference in this case to have you fork it rather then add the ifdef stuff. But if you feel really strongly not too we would hear you out.

Yes we really prefer not to for the runner.

JacobSzwejbka reviewed (approved):
Ok that works for me

JacobSzwejbka committed:
Merge branch 'main' into vgf
